common/views/components/IIIFViewer/IIIFViewer.js

```diff
@@ -0,0 +1,339 @@
+// @flow
+import styled from 'styled-components';
+import { classNames, spacing, font } from '@weco/common/utils/classnames';
+import NextLink from 'next/link';
+import { itemUrl } from '@weco/common/services/catalogue/urls';
+import {
+  type IIIFCanvas,
+  type IIIFImageService,
+} from '@weco/common/model/iiif';
+import Paginator, {
+  type PaginatorRenderFunctionProps,
+  type PropsWithoutRenderFunction as PaginatorPropsWithoutRenderFunction,
+} from '@weco/common/views/components/RenderlessPaginator/RenderlessPaginator';
+import Control from '@weco/common/views/components/Buttons/Control/Control';
+import IIIFResponsiveImage from '@weco/common/views/components/IIIFResponsiveImage/IIIFResponsiveImage';
+
+const IIIFViewerPaginatorButtons = styled.div.attrs(props => ({
+  className: classNames({
+    'flex absolute flex--h-center': true,
+  }),
+}))`
+  left: ${props => (props.isThumbs ? 'auto' : '50%')};
+  right: ${props => props.theme.spacingUnit}px;
+  bottom: ${props => (props.isThumbs ? '50%' : props.theme.spacingUnit + 'px')};
+  transform: ${props =>
+    props.isThumbs ? 'translateY(50%)' : 'translateX(-50%)'};
+
+  ${props =>
+    props.isThumbs &&
+    `
+    @media (min-width: ${props.theme.sizes.medium}px) {
+      bottom: ${props.theme.spacingUnit}px;
+      left: 50%;
+      transform: translateX(-50%) translateY(0%);
+
+      .control__inner {
+        transform: rotate(90deg);
+      }
+    }
+  `}
+`;
+
+const IIIFViewerThumbNumber = styled.span.attrs(props => ({
+  className: classNames({
+    'line-height-1': true,
+    'absolute bg-charcoal font-white': true,
+    [font({ s: 'LR3' })]: true,
+  }),
+}))`
+  top: ${props => props.theme.spacingUnit}px;
+  left: 50%;
+  transform: translateX(-50%);
+  padding: 3px 2px 0;
+  z-index: 1;
+`;
+
+const IIIFViewerThumb = styled.div.attrs(props => ({
+  className: classNames({
+    'relative flex flex--v-center': true,
+    [spacing({ s: 1 }, { padding: ['top', 'right', 'bottom', 'left'] })]: true,
+  }),
+}))`
+  height: 100%;
+  width: 20%;
+  margin-right: ${props => props.theme.spacingUnit}px;
+
+  &:last-child {
+    margin: 0;
+  }
+
+  @media (min-width: ${props => props.theme.sizes.medium}px) {
+    height: 25%;
+    width: 100%;
+    margin-right: 0;
+  }
+`;
+
+const IIIFViewerThumbs = styled.div.attrs(props => ({
+  className: classNames({
+    'flex flex--h-center relative bg-smoke': true,
+  }),
+}))`
+  height: 20%;
+  width: 100%;
+  padding: 0 100px 0 0;
+
+  @media (min-width: ${props => props.theme.sizes.medium}px) {
+    height: 100%;
+    flex-direction: column;
+    width: 25%;
+    padding: 0 0 ${props => props.theme.spacingUnit * 10}px;
```

Comment:
repeat: spacing classes?

Reply:
I'd like to avoid spending too much time on (semantics of) styling for this since it hasn't been designed by a UI designer and is almost certainly going to undergo drastic change (it's in the WIP directory in Cardigan). The work done here was just what's required to get the MVP out.